pilot/model/validation.go

```diff
@@ -1805,8 +1919,13 @@ func validateDestination(destination *routingv2.Destination) (errs error) {
 		return
 	}
 
-	errs = appendErrors(errs, Labels(destination.Labels).Validate())
-	// TODO: Name
+	// TODO: validate short name / FQDN
+	if destination.Name == "" {
+		errs = appendErrors(errs, fmt.Errorf("Route Rule destination should not be empty"))
+	}
+
+	// TODO: Destination Subset string? invalid characters?
```

Comment:
what subset string?

Reply:
This one:
https://github.com/istio/api/blob/master/routing/v1alpha2/route_rule.proto#L284

It is optional, so the validation would be: if it is non-empty, make sure that the subset name is valid.